crates/subspace-farmer/src/utils/farmer_piece_getter.rs

```diff
@@ -32,6 +34,7 @@ struct Inner<PV, NC> {
     node_client: NC,
     plotted_pieces: Arc<Mutex<Option<PlottedPieces>>>,
     dsn_cache_retry_policy: DsnCacheRetryPolicy,
+    in_progress_pieces: Mutex<HashMap<PieceIndex, Arc<AsyncMutex<Option<Piece>>>>>,
```

Comment:
Why do you use both Mutex and AsyncMutex here?

Reply:
`Mutex` is generally faster and preferred, but I do need to hold guard of `AsyncMutex` across `.await`, hence async mutex is needed there.